dev-libs/openssl/openssl3-3.0.13.recipe

```diff
@@ -106,7 +106,7 @@ defineDebugInfoPackage openssl3$secondaryArchSuffix \
 BUILD()
 {
 	./config --prefix=$prefix --libdir=$relativeLibDir \
-		--openssldir=$dataRootDir/ssl \
+		--openssldir=$dataRootDir/ssl3 \
```

Comment:
probably a bad idea, because this is where the root certificates are located. Does it work nonetheless?

Reply:
Seems to fixed the earlier warning on libproviders.so:
```
/Opslag/haikuports> pkgman install qt_creator
100% repochecksum-1 [65 bytes]
Validating checksum for Haiku...done.
100% repochecksum-1 [64 bytes]
Validating checksum for HaikuPorts...done.
The following changes will be made:
  in system:
    install package qt_creator-13.0.0-1 from repository HaikuPorts
Continue? [yes/no] (yes) : 
Re-using download '/boot/system/packages/administrative/state_2024-05-02_14:24:55/qt_creator-13.0.0-1-x86_64.hpkg' from previous transaction
Validating checksum for https://eu.hpkg.haiku-os.org/haikuports/r1beta4/x86_64/current/packages/qt_creator-13.0.0-1-x86_64.hpkg...done.
[system] Applying changes ...
```
just checked on 64bit R1B4 on a clean build/install for openssl3

Reply:
I mean: does openssl3 work?

Reply:
If I knew how to check this I could answer that :) Just did a build for a 64bit anyboot iso with these installed, as far as it should use ssl_devel it can only pick up openssl3 as openssl_devel conflicts/isn't installed here atm.
Build went fine and image boots fine in qemu.
At this moment all I could do for now is build/install the packages, see if nothing is broken.
If there is a quick way to check it I'm all ears. :)

Reply:
OK, not sure if it's correct, but getting this output in Terminal:
```
~/Desktop> openssl_3 s_client -connect example.com:443
CONNECTED(00000004)
depth=1 C = US, O = DigiCert Inc, CN = DigiCert Global G2 TLS RSA SHA256 2020 CA1
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 C = US, ST = California, L = Los Angeles, O = Internet\C2\A0Corporation\C2\A0for\C2\A0Assigned\C2\A0Names\C2\A0and\C2\A0Numbers, CN = www.example.org
verify return:1
---
Certificate chain
 0 s:C = US, ST = California, L = Los Angeles, O = Internet\C2\A0Corporation\C2\A0for\C2\A0Assigned\C2\A0Names\C2\A0and\C2\A0Numbers, CN = www.example.org
   i:C = US, O = DigiCert Inc, CN = DigiCert Global G2 TLS RSA SHA256 2020 CA1
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Jan 30 00:00:00 2024 GMT; NotAfter: Mar  1 23:59:59 2025 GMT
 1 s:C = US, O = DigiCert Inc, CN = DigiCert Global G2 TLS RSA SHA256 2020 CA1
   i:C = US, O = DigiCert Inc, OU = www.digicert.com, CN = DigiCert Global Root G2
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Mar 30 00:00:00 2021 GMT; NotAfter: Mar 29 23:59:59 2031 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIHbjCCBlagAwIBAgIQB1vO8waJyK3fE+Ua9K/hhzANBgkqhkiG9w0BAQsFADBZ
MQswCQYDVQQGEwJVUzEVMBMGA1UEChMMRGlnaUNlcnQgSW5jMTMwMQYDVQQDEypE
aWdpQ2VydCBHbG9iYWwgRzIgVExTIFJTQSBTSEEyNTYgMjAyMCBDQTEwHhcNMjQw
MTMwMDAwMDAwWhcNMjUwMzAxMjM1OTU5WjCBljELMAkGA1UEBhMCVVMxEzARBgNV
BAgTCkNhbGlmb3JuaWExFDASBgNVBAcTC0xvcyBBbmdlbGVzMUIwQAYDVQQKDDlJ
bnRlcm5ldMKgQ29ycG9yYXRpb27CoGZvcsKgQXNzaWduZWTCoE5hbWVzwqBhbmTC
oE51bWJlcnMxGDAWBgNVBAMTD3d3dy5leGFtcGxlLm9yZzCCASIwDQYJKoZIhvcN
AQEBBQADggEPADCCAQoCggEBAIaFD7sO+cpf2fXgCjIsM9mqDgcpqC8IrXi9wga/
9y0rpqcnPVOmTMNLsid3INbBVEm4CNr5cKlh9rJJnWlX2vttJDRyLkfwBD+dsVvi
vGYxWTLmqX6/1LDUZPVrynv/cltemtg/1Aay88jcj2ZaRoRmqBgVeacIzgU8+zmJ
7236TnFSe7fkoKSclsBhPaQKcE3Djs1uszJs8sdECQTdoFX9I6UgeLKFXtg7rRf/
hcW5dI0zubhXbrW8aWXbCzySVZn0c7RkJMpnTCiZzNxnPXnHFpwr5quqqjVyN/aB
KkjoP04Zmr+eRqoyk/+lslq0sS8eaYSSHbC5ja/yMWyVhvMCAwEAAaOCA/IwggPu
MB8GA1UdIwQYMBaAFHSFgMBmx9833s+9KTeqAx2+7c0XMB0GA1UdDgQWBBRM/tAS
TS4hz2v68vK4TEkCHTGRijCBgQYDVR0RBHoweIIPd3d3LmV4YW1wbGUub3Jnggtl
eGFtcGxlLm5ldIILZXhhbXBsZS5lZHWCC2V4YW1wbGUuY29tggtleGFtcGxlLm9y
Z4IPd3d3LmV4YW1wbGUuY29tgg93d3cuZXhhbXBsZS5lZHWCD3d3dy5leGFtcGxl
Lm5ldDA+BgNVHSAENzA1MDMGBmeBDAECAjApMCcGCCsGAQUFBwIBFhtodHRwOi8v
d3d3LmRpZ2ljZXJ0LmNvbS9DUFMwDgYDVR0PAQH/BAQDAgWgMB0GA1UdJQQWMBQG
CCsGAQUFBwMBBggrBgEFBQcDAjCBnwYDVR0fBIGXMIGUMEigRqBEhkJodHRwOi8v
Y3JsMy5kaWdpY2VydC5jb20vRGlnaUNlcnRHbG9iYWxHMlRMU1JTQVNIQTI1NjIw
MjBDQTEtMS5jcmwwSKBGoESGQmh0dHA6Ly9jcmw0LmRpZ2ljZXJ0LmNvbS9EaWdp
Q2VydEdsb2JhbEcyVExTUlNBU0hBMjU2MjAyMENBMS0xLmNybDCBhwYIKwYBBQUH
AQEEezB5MCQGCCsGAQUFBzABhhhodHRwOi8vb2NzcC5kaWdpY2VydC5jb20wUQYI
KwYBBQUHMAKGRWh0dHA6Ly9jYWNlcnRzLmRpZ2ljZXJ0LmNvbS9EaWdpQ2VydEds
b2JhbEcyVExTUlNBU0hBMjU2MjAyMENBMS0xLmNydDAMBgNVHRMBAf8EAjAAMIIB
fQYKKwYBBAHWeQIEAgSCAW0EggFpAWcAdABOdaMnXJoQwzhbbNTfP1LrHfDgjhuN
acCx+mSxYpo53wAAAY1b0vxkAAAEAwBFMEMCH0BRCgxPbBBVxhcWZ26a8JCe83P1
JZ6wmv56GsVcyMACIDgpMbEo5HJITTRPnoyT4mG8cLrWjEvhchUdEcWUuk1TAHYA
fVkeEuF4KnscYWd8Xv340IdcFKBOlZ65Ay/ZDowuebgAAAGNW9L8MAAABAMARzBF
AiBdv5Z3pZFbfgoM3tGpCTM3ZxBMQsxBRSdTS6d8d2NAcwIhALLoCT9mTMN9OyFz
IBV5MkXVLyuTf2OAzAOa7d8x2H6XAHcA5tIxY0B3jMEQQQbXcbnOwdJA9paEhvu6
hzId/R43jlAAAAGNW9L8XwAABAMASDBGAiEA4Koh/VizdQU1tjZ2E2VGgWSXXkwn
QmiYhmAeKcVLHeACIQD7JIGFsdGol7kss2pe4lYrCgPVc+iGZkuqnj26hqhr0TAN
BgkqhkiG9w0BAQsFAAOCAQEABOFuAj4N4yNG9OOWNQWTNSICC4Rd4nOG1HRP/Bsn
rz7KrcPORtb6D+Jx+Q0amhO31QhIvVBYs14gY4Ypyj7MzHgm4VmPXcqLvEkxb2G9
Qv9hYuEiNSQmm1fr5QAN/0AzbEbCM3cImLJ69kP5bUjfv/76KB57is8tYf9sh5ik
LGKauxCM/zRIcGa3bXLDafk5S2g5Vr2hs230d/NGW1wZrE+zdGuMxfGJzJP+DAFv
iBfcQnFg4+1zMEKcqS87oniOyG+60RMM0MdejBD7AS43m9us96Gsun/4kufLQUTI
FfnzxLutUV++3seshgefQOy5C/ayi8y1VTNmujPCxPCi6Q==
-----END CERTIFICATE-----
subject=C = US, ST = California, L = Los Angeles, O = Internet\C2\A0Corporation\C2\A0for\C2\A0Assigned\C2\A0Names\C2\A0and\C2\A0Numbers, CN = www.example.org
issuer=C = US, O = DigiCert Inc, CN = DigiCert Global G2 TLS RSA SHA256 2020 CA1
---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: RSA-PSS
Server Temp Key: ECDH, prime256v1, 256 bits
---
SSL handshake has read 3821 bytes and written 747 bytes
Verification error: unable to get local issuer certificate
---
New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
Server public key is 2048 bit
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 20 (unable to get local issuer certificate)
---
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: F4F1D56AF391AA4DB99E52DBCC7566695A9B4C682B6CE8C4734B8694E00E01A0
    Session-ID-ctx: 
    Resumption PSK: 5FCEFDC3C04F67A4751637A3EF4A68657E256720E6F3D0812EC822CE367A311663FF117551833C610D855FA66EC21A4A
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - 6e e8 7c 10 e1 ec 9d fc-ba e9 28 42 c0 ad c3 5a   n.|.......(B...Z
    0010 - 9a a9 62 72 69 cb 58 6f-3b cd b5 11 ea 84 bf b7   ..bri.Xo;.......
    0020 - eb 7b b6 11 23 bf 4e c8-4d 9e 78 b7 05 97 34 df   .{..#.N.M.x...4.
    0030 - ed 87 1a 32 f8 4c 83 90-79 b9 9e 2d 11 7a e0 05   ...2.L..y..-.z..
    0040 - 93 55 3f 85 b3 6e 8a c7-79 cc 8d 60 7b 86 2f 1d   .U?..n..y..`{./.
    0050 - 0e 72 8e 1e 9a 5c 1f e2-70 e1 58 74 d0 ce 56 46   .r...\..p.Xt..VF
    0060 - fc cf 31 ae 09 2c 28 4c-96 66 e2 ca 58 50 2d 42   ..1..,(L.f..XP-B
    0070 - 35 87 76 f3 11 6c ef be-52 8e 00 2f c2 16 4b 08   5.v..l..R../..K.
    0080 - 2d 0d ee bb 8e 2c 4a 8a-bc d1 6b b3 08 91 07 99   -....,J...k.....
    0090 - e2 09 67 88 cb cf 25 6f-c6 c3 6a eb c4 45 7d f6   ..g...%o..j..E}.
    00a0 - 4b 30 90 80 cf 07 44 03-56 94 7a 27 2f ec c1 4f   K0....D.V.z'/..O
    00b0 - eb c3 28 8e c9 b5 7b 09-89 c0 bc 9f 8c 8b 34 26   ..(...{.......4&

    Start Time: 1717503349
    Timeout   : 7200 (sec)
    Verify return code: 20 (unable to get local issuer certificate)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: 64BA67430DBE8B5D716C57E0828CA708A4CAD441313ED370D0821FC5CE2603AD
    Session-ID-ctx: 
    Resumption PSK: 6A568BBCBB3AC5677209E6FB0E475E28E8B8CD5E8B3B7B9F81EF98DA648E6778BB8D8950604B37EE801A37BA1502B0C7
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - 6e e8 7c 10 e1 ec 9d fc-ba e9 28 42 c0 ad c3 5a   n.|.......(B...Z
    0010 - cf 81 e4 49 8f 51 3c 1c-b3 a7 09 11 9a ac ad d4   ...I.Q<.........
    0020 - 2a 47 06 44 9b f3 7f e6-be ee d4 39 fa 2c 83 7f   *G.D.......9.,..
    0030 - ba 38 63 f2 16 28 69 9e-a5 2e 5e 86 79 86 5f 99   .8c..(i...^.y._.
    0040 - 6b 7e 38 43 bc 34 d0 46-c8 4a 60 99 81 0c 38 53   k~8C.4.F.J`...8S
    0050 - 3c fb 03 fd 82 0e 34 a9-d0 fb 9d 7a 9b f7 41 89   <.....4....z..A.
    0060 - 44 01 3c e0 c2 b4 f7 a2-d4 97 93 80 2d 4a e2 48   D.<.........-J.H
    0070 - 96 61 98 35 ae a5 e2 0d-13 6b d9 61 9d a2 62 70   .a.5.....k.a..bp
    0080 - 88 0a 4a 25 8a 29 cd 0e-fb a8 79 47 47 50 63 d7   ..J%.)....yGGPc.
    0090 - 5f cc 54 bb 62 ea 8b 44-c1 5e 5c 6e 76 0e b7 c1   _.T.b..D.^\nv...
    00a0 - 44 ec 73 29 32 e5 6a e3-f6 1e 25 75 41 c4 33 0e   D.s)2.j...%uA.3.
    00b0 - d4 75 dc 31 71 f7 f4 de-81 aa cd 53 8e 9c 33 5b   .u.1q......S..3[

    Start Time: 1717503349
    Timeout   : 7200 (sec)
    Verify return code: 20 (unable to get local issuer certificate)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
bad select -2147483638
~/Desktop> 
```

Reply:
Ah, I see my previous reply got mangled because there was an @ in an URL, great. Here is a link in a different archive of the same mailing list... https://www.spinics.net/lists/openssl-users/msg15473.html

I changed this to fix the mentionned issue about libproviders.so.

> verify error:num=20:unable to get local issuer certificate

This means it didn't find the certificates. So, yes, that will be a problem as more things start using OpenSSL 3. Good catch :)

Maybe we can add a link to the certificates in the other directory, or change something in the configuration file, so that the certificates are still found in the previous location. But we need a different directory for other configuration files because they are not compatible (unless we give up on having both versions co-exist).

Reply:
"unable to get local issuer certificate" doesn't sound good

Reply:
Not sure if it's the right way (thinking out loud), https://sources.debian.org/patches/openssl/3.3.0-1/Remove-the-provider-section.patch/
With this and --openssldir=$dataRootDir/ssl I now get:
```
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: C21940C1EACFD36488CE472AB1A068B502B05AF4799C76F6D83A80B62EA958C6
    Session-ID-ctx: 
    Resumption PSK: 8A7F9CEBCD68C55770E3131D4C597FD344D30ABAC7BABD081C8997CBCB2730D3A56CCBCCE80C8804C67A4EB635314517
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - 47 e9 ed c1 e0 7c 51 24-3d ca 45 ba 10 75 b2 d8   G....|Q$=.E..u..
    0010 - 39 92 80 2d 03 e4 5a 8c-24 f5 99 8f 76 42 86 2a   9..-..Z.$...vB.*
    0020 - 97 c4 66 e8 bd f5 40 62-51 3b f3 e1 ce 80 42 e9   ..f...@bQ;....B.
    0030 - 32 17 08 b0 81 7a 9f ec-73 cc 8e ff c5 ec d8 fb   2....z..s.......
    0040 - 45 3e 61 3f 16 c9 ff c9-1b 33 4d 1b ab 19 50 a6   E>a?.....3M...P.
    0050 - d8 38 e9 6a 06 bc 39 51-c4 17 7d ce e9 1f 44 e0   .8.j..9Q..}...D.
    0060 - 81 19 79 d6 f6 19 ef 35-02 9f 65 7c 04 4b 00 68   ..y....5..e|.K.h
    0070 - 18 4f 67 fa 93 ed 15 5d-55 62 c5 f2 ee 88 ec 9c   .Og....]Ub......
    0080 - ef 4b 89 a9 e1 f4 81 20-13 5d 20 01 02 a2 59 8c   .K..... .] ...Y.
    0090 - 66 93 5d fd cb c2 01 8a-b1 6e c7 18 75 d8 c0 1b   f.]......n..u...
    00a0 - 6b 4b 04 fa d4 62 8e ba-92 b0 ac 3d 68 f5 fe cf   kK...b.....=h...
    00b0 - 7f f6 c3 7a 8b 8a aa 9a-c1 1b 1e 24 c2 82 90 8d   ...z.......$....
    00c0 - 84 f3 6e 81 42 60 36 af-16 78 d7 68 1f 33 a1 2c   ..n.B`6..x.h.3.,

    Start Time: 1717506496
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
qHTTP/1.0 501 Not Implemented
Content-Type: text/html
Content-Length: 357
Connection: close
Date: Tue, 04 Jun 2024 13:08:55 GMT
Server: ECLF (dcd/7D84)
```

Reply:
Yes, I think that would work, no need for a separate directory then. We can share the same ssl/ directory between the two versions